kong/plugins/oauth2/secret.lua

```diff
@@ -0,0 +1,159 @@
+local utils = require "kong.tools.utils"
+
+
+local type = type
+local find = string.find
+local pcall = pcall
+local remove = table.remove
+local concat = table.concat
+local assert = assert
+
+
+--local CORES
+--do
+--  local infos = utils.get_system_infos()
+--  if type(infos) == "table" then
+--    CORES = infos.cores
+--  end
+--  if not CORES then
+--    CORES = ngx.worker.count() or 1
+--  end
+--end
+
+
+local PREFIX = nil -- currently chosen algorithm
+
+
+local ARGON2
+local ARGON2_ID = "$argon2"
+do
+  local ARGON2_PREFIX
+  local ok, crypt = pcall(function()
```

Comment:
I suggest not wrapping the whole declaration with `pcall`. Consider what will happen if someone modifies this code in the future and introduces a typo which makes this function always fail. Then the app will always pick bcrypt. If the `pcall` is supposed to detect that `require "argon2"` throws, then wrap only that with `pcall`.

Reply:
AFAIK `require "argon"` might not fail, and you never know if someone decides to update dependency and it fails in different place?